network/basics/get_url.py

```diff
@@ -212,6 +220,26 @@ def extract_filename_from_headers(headers):
 
     return res
 
+def get_available_hash_algorithms():
+    available_hash_algorithms = dict()
+    try:
+        import hashlib
+
+        # python 2.7.9+ and 2.7.0+
+        for attribute in ('available_algorithms', 'algorithms'):
+            algorithms = getattr(hashlib, attribute, None)
+        if algorithms is None:
+            # python 2.5+
+            algorithms = ('md5', 'sha1', 'sha224', 'sha256', 'sha384', 'sha512')
+        for algorithm in algorithms:
+            available_hash_algorithms[algorithm] = getattr(hashlib, algorithm)
+    except ImportError:
+        import md5
+        import sha
+        available_hash_algorithms = {'md5': md5.md5,
+                                     'sha1': sha.sha}
+    return available_hash_algorithms
+
```

Comment:
nice.  I like this.  You're right we might want to move this into the module_utils/* code at some point in the future.